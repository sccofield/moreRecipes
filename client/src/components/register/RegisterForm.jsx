import React from 'react';
import PropTypes from 'prop-types';

const RegisterForm = props => (
  <div className="main">
    <div className="row justify-content-md-center">
      <div className="col-md-8 col-lg-6 col-12">
        <div className="registeration text-center">
          <h2>Register</h2>

          <div className="form-header">
            <p>Already have an account?
              <span>Sign in</span>
            </p>
          </div>
          {
            props.errorMessage &&
            <p className="errorMessage">{props.errorMessage}</p>
          }
          {
            props.state.errors &&
            <ul className="errorMessage">{
              props.state.errors.map(error => <li key={Math.floor(Math.random() * 1000)} >{error}</li>)
            }
            </ul>}
          <form onSubmit={props.onSubmit}>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                name="userName"
                placeholder="Name"
                value={props.state.userName}
                onChange={props.onChange}
                required
              />
            </div>
            <div className="form-group">
              <input
                type="email"
                name="email"
                className="form-control"
                id="exampleInputEmail1"
                aria-describedby="emailHelp"
                placeholder="Email"
                value={props.state.email}
                onChange={props.onChange}
                required
              />
            </div>

            <div className="form-group">
              <input
                type="password"
                name="password"
                className="form-control"
                id="exampleInputPassword1"
                placeholder="Password"
                value={props.state.password}
                onChange={props.onChange}
                required
              />
            </div>

            <div className="form-group">
              <input
                type="password"
                name="cPassword"
                className="form-control"
                placeholder="Re-enter Password"
                value={props.state.cPassword}
                onChange={props.onChange}
                required
              />
            </div>
            <button
              type="submit"
              id="registerSubmit"
              className="btn btn-primary recipeButton"
            >Create Account
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
);

RegisterForm.propTypes = {
  errorMessage: PropTypes.string,
  errors: PropTypes.arrayOf(PropTypes.string),
  state: PropTypes.shape({
    cPassword: PropTypes.string,
    password: PropTypes.string,
    userName: PropTypes.string,
    errors: PropTypes.array,
    email: PropTypes.string
  }).isRequired,
  onChange: PropTypes.func,
  onSubmit: PropTypes.func
};
RegisterForm.defaultProps = {
  errorMessage: null,
  errors: null,
  onChange: null,
  onSubmit: null
};
export default RegisterForm;
