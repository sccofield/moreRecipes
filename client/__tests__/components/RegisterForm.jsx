import React from 'react';
import toJSON from 'enzyme-to-json';
import RegisterForm from '../../src/components/register/RegisterForm';

describe('Recipe component', () => {
  const props = {
    errorMessage: null,
    errors: null,
    state: {
      cPassword: 'password',
      password: 'password',
      userName: 'test',
      errors: null,
      email: 'test@test.com'
    },
    
  };

  const errorProps = {
    errorMessage: null,
    errors: null,
    state: {
      cPassword: 'password',
      password: 'password',
      userName: 'test',
      errors: ['new error'],
      email: 'test@test.com'
  },
  
}

const errorProps2 = {
  errorMessage: 'error',
  errors: null,
  state: {
    cPassword: 'password',
    password: 'password',
    userName: 'test',
    errors: null,
    email: 'test@test.com'
},

}

  it('should render register form when props is given', () => {
    const wrapper = shallow(<RegisterForm {...props} />);
    expect(wrapper.find('div').length).toEqual(9);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should show error message when there is an error in state', ()=> {
    const wrapper = shallow(<RegisterForm {...errorProps } />);
    expect(wrapper.find('.errorMessage').text()).toBe('new error');
    expect(wrapper.find('.errorMessage li').length).toEqual(1);
    
  });

  it('should show error message when there is an error in props', ()=> {
    const wrapper = shallow(<RegisterForm {...errorProps2 } />);
    expect(wrapper.find('.errorMessage').text()).toBe('error');
    expect(wrapper.find('.errorMessage').length).toEqual(1);
    
  })

});

