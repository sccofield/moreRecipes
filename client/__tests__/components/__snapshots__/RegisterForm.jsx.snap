// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component should render register form when props is given 1`] = `
<div
  className="main"
>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="col-md-8 col-lg-6 col-12"
    >
      <div
        className="registeration text-center"
      >
        <h2>
          Register
        </h2>
        <div
          className="form-header"
        >
          <p>
            Already have an account?
            <span>
              Sign in
            </span>
          </p>
        </div>
        <form
          onSubmit={null}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="userName"
              onChange={null}
              placeholder="Name"
              required={true}
              type="text"
              value="test"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              aria-describedby="emailHelp"
              className="form-control"
              id="exampleInputEmail1"
              name="email"
              onChange={null}
              placeholder="Email"
              required={true}
              type="email"
              value="test@test.com"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="exampleInputPassword1"
              name="password"
              onChange={null}
              placeholder="Password"
              required={true}
              type="password"
              value="password"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="cPassword"
              onChange={null}
              placeholder="Re-enter Password"
              required={true}
              type="password"
              value="password"
            />
          </div>
          <button
            className="btn btn-primary recipeButton"
            id="registerSubmit"
            type="submit"
          >
            Create Account
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
